File: OMSWeb/oms-track-monitor/src/zcu/types/Zcu.ts
import { ZcuCompletePoint } from "./ZcuCompletePoint";
import { ZcuInputZone } from "./ZcuInputZone";

type Zcu = {
  id: number
  x: number
  y: number
  usingType:
  | 0 // none
  | 1 // hw
  | 2 // sw
  error: boolean
  zcuType: number
  inputZones: ZcuInputZone[]
  completePoints: ZcuCompletePoint[]
}

export { Zcu }

File: OMSWeb/Models/Tracks/Zcu.cs
namespace OMSWeb.Models.Tracks
{
    public class Zcu
    {
        public int Id { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
        public int usingType { get; set; }
        public int ZcuType { get; set; }
        public ZcuCompletePoint[] CompletePoints { get; set; }
        public ZcuInputZone[] InputZones { get; set; }
        public bool error { get; set; }
    }

    public class ZcuCompletePoint
    {
        public int Id { get; set; }
        public int ZcuId { get; set; }
        public int CompletePointId { get; set; }
    }

    public class ZcuInputZone
    {
        public int Id { get; set; }
        public int ZcuId { get; set; }
        public int PriorityPoint { get; set; }
        public string ZonePoints { get; set; }
    }

    public class ZcuStatus : Entities.ZcuStatusEntity
    {
    }
}